### PR TITLE
non-DMA SPI working; adding this now for testing; will continue with DMA

### PR DESCRIPTION
**Pull request by dhalbert**
Tested with BMP280 SPI on '21 and '51

needs corresponding ASF4 PR as well

Also, fixed pin mappings for rev B Metro M4:
swap PA12 and PA13 on SPI 2x3 header
swap A3 and A5

Comment out all frozen modules in CPX again to make room while waiting
for SPI flash.

### CODE REVIEW

File: ports/atmel-samd/common-hal/busio/SPI.c [hunk 1]
```diff
@@ -25,25 +25,35 @@
...
-// We use ENABLE registers below we don't want to treat as a macro.
-#undef ENABLE
+#include "hpl_sercom_config.h"
+#include "peripheral_clk_config.h"
+
+#include "hal/include/hal_gpio.h"
+#include "hal/include/hal_spi_m_sync.h"
+#include "hal/include/hpl_spi_m_sync.h"
+
+#include "peripherals.h"
```
Comment (tannewt) [inline ports/atmel-samd/common-hal/busio/SPI.c]: This change has two _peripherals.h files. Did you mean to merge them?

Comment (dhalbert) [reply ports/atmel-samd/common-hal/busio/SPI.c]: Hi - `peripherals.h` includes either `samd21_peripherals.h` or `samd51_peripherals.h`. They are slightly different. Is that what you mean?

Comment (tannewt) [reply ports/atmel-samd/common-hal/busio/SPI.c]: I don't see `peripherals.h` in the diff.

Comment (dhalbert) [reply ports/atmel-samd/common-hal/busio/SPI.c]: I added it in the previous PR when I did I2C. It hasn't changed: it's just conditional `#include`s.

**Comment by tannewt** [comment]
Ah ok! You'll need to fix the build because CircuitPlayground is out of space.

**Pull request merged**